### ISSUE TITLE
ToC in pages mode

### ISSUE THREAD

**Issue by enketo-issue-mover** [OpenClinica/enketo-express-oc#49]
_From @MartijnR on December 1, 2017 21:47_

Requirements still under discussion.

Label could be in the first label (group or question) on the page.

* ToC accessible on all pages or just first page?
* behavior with pageValidation: true
* behavior with pageValidation: false, when error message is just being shown (might not actually be relevant if the ToC is a pulldown menu as it will take time between clicking the pulldown and clicking the page, sufficient to see the error).
* OC only behavior with RFC fields

_Copied from original issue: kobotoolbox/enketo-express#887_

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on February 26, 2018 18:4_

* behavior with translated forms (update ToC when language switches)
* ToC updates when page becomes irrelevant - grey out?
* ToC updates when question with first page label becomes irrelevant but not the rest of the page

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on July 27, 2018 19:54_

just lay the groundwork to build a ToC and flip to a page

skipping may not be an issue as long as the current page allows going to next

if label becomes irrelevant, swap to next label according to page-naming-algorithm

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on July 30, 2018 22:22_

* [x] figure out why pages module initializes twice
* [x] add tests
* [x] fix first page shown (it starts on page 5 !?) -> this is caused by removing second pages.init()
* [x] fix failing test wrt relevant
* [x] fix language switch

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on August 3, 2018 20:1_

* [ ] figure out how best to suppress a ToC (form-wide (API, or style?)) (Also check ODK Collect)
* [x] create the UI
* [x] create the logic
* [x] test with repeats as page
* [x] augment enketo-core ui in enketo-express
* [ ] same in enketo-express-oc

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on August 9, 2018 19:22_

* [x] small screen
* [x] large screen
* [x] with branding 
* [x] without branding
* [x] with languages
* [x] without languages
* [x] with print button
* [x] without print button
* [x] with mobile bookmark button

fix this:
<img width="596" alt="screen shot 2018-08-09 at 3 25 51 pm" src="https://user-images.githubusercontent.com/627350/43926773-927b2b44-9be8-11e8-992d-8bb860799d3b.png">
<img width="437" alt="screen shot 2018-08-09 at 3 25 42 pm" src="https://user-images.githubusercontent.com/627350/43926775-93d67746-9be8-11e8-99ba-a9f6f4b4a023.png">

remove the separator:
<img width="736" alt="screen shot 2018-08-09 at 3 25 13 pm" src="https://user-images.githubusercontent.com/627350/43926783-95bf7e04-9be8-11e8-950d-646f922f0a73.png">

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on August 9, 2018 21:57_

<img width="859" alt="screen shot 2018-08-09 at 3 53 28 pm" src="https://user-images.githubusercontent.com/627350/43928099-f0099f12-9bec-11e8-9314-ab49e433ccac.png">
<img width="716" alt="screen shot 2018-08-09 at 3 53 23 pm" src="https://user-images.githubusercontent.com/627350/43928101-f2a8c82e-9bec-11e8-81ab-63f7d1a27d33.png">

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on August 9, 2018 22:18_

https://forum.opendatakit.org/t/blocking-page-jumps/14500

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on August 10, 2018 16:1_

When blocking jumps, we should probably also block GoToFirst and GoToLast.

**Comment by enketo-issue-mover** [comment]
_From @MartijnR on August 10, 2018 19:17_


<img width="892" alt="screen shot 2018-08-10 at 1 17 59 pm" src="https://user-images.githubusercontent.com/627350/43976965-d5d70a54-9c9f-11e8-90b9-defbd6c9749c.png">

**Comment by pbowen-oc** [comment]
I think this was ready for our review

**Comment by MartijnR** [comment]
yes, but note that the latest Enketo Core has completely changed this feature. So you may want to wait until that is merged into enketo-express-oc

**Comment by pbowen-oc** [comment]
Would that be part of the next merge to our master branch?
